text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+The Sigstore community has written a [document](https://docs.google.com/document/d/1mXrVAkUA9dd4M7fa_AJC8mQ55YnYJ-DKsGq30lh0FvA/edit#) with recommendations and alternative designs for projects integrating Sigstore with artifact repositories.
+
+## Articles and papers
+
+The [Sigstore Blog](https://blog.sigstore.dev/) contains a lot of articles related Sigstore. In particular, [this post](https://blog.sigstore.dev/signatus-ergo-securus-who-can-sign-what-with-tuf-and-sigstore-ea4d3d84b8b6) covers the problems faced by similar package systems like PyPI.
+
+The GitHub tema has published a few [blog posts](https://github.blog/2022-10-25-why-were-excited-about-the-sigstore-general-availability/) on Sigstore. 
+
+A [paper](https://dl.acm.org/doi/abs/10.1145/3548606.3560596) about Sigstore on ACM.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+* Opt-in or opt-out of signing and/or verification? Making it opt-out could increase the percentage of signed crates, but making it opt-in to start with seems like the safest choice.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
+
+* Use the The Update Framework (TUF) to manage trust roots for their own Sigstore instance, which would make it independent of the public Sigstore instances. 
+* Use the public Sigstore instance to manage root keys for TUF.
+* Extending other cargo commands to make use of the information stored in the transparency log when listing dependencies and other places where it makes sense.
+* Integrating with [in-toto.io](https://in-toto.io) to provide more advanced attestation of artifacts.
+* Build source attestation: adding special modes for public CI systems such as GitHub actions where trusted builders can sign to attest that artifact is built from a particular source.
+
+* Use the public Sigstore instance to manage root keys for TUF.
+* Extending other cargo commands to make use of the information stored in the transparency log when listing dependencies and other places where it makes sense.
+* Integrating with [in-toto.io](https://in-toto.io) to provide more advanced attestation of artifacts..
+
+# Threat model
+
+The threat model is based on the [NPM RFC](https://github.com/npm/rfcs/blob/main/accepted/0049-link-packages-to-source-and-build.md).
+
+Our threat model focuses on a package hijacking attack where an attacker gains access to a crate maintainer's crates.io credentials and uploads a modified version of the crate under a version number that is likely to get pulled in by cargo. Developers that use this crate cannot know whether the crate is built from the original source repository or on the attacker's laptop.
+
+This proposal does not mitigate against compromised crates.io accounts. The aim is to make it harder to execute these types of attacks by creating a public audit trail for where, how and who published a package. Over time the presence of this information can be enforced.
```

Comment:
It might be better to say that this doesn't mitigate against compromised *GitHub* accounts.  At least today, a crates.io account could also be compromised by stealing a crates.io API token, and authenticating with GitHub on every crate publish would mitigate against that.